pkg/backporter/backporter.go

```diff
@@ -278,94 +304,218 @@ func GetBugHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("Bug#%d Details of parent could not be retrieved : %v", bug.ID, err))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return nil, wpErr
+		}
+		return nil, err
+	}
+
+	clones, err := client.GetAllClones(bug)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		wpErr := writePage(w, "Not Found", errorTemplate, err.Error())
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
+			return nil, wpErr
+		}
+		return nil, err
+	}
+	// Target versions would be used to populate the CreateClone dropdown
+	targetVersions := sets.NewString(allTargetVersions.List()...)
+	// Remove target versions of the original bug
+	for _, release := range bug.TargetRelease {
+		_, ok := targetVersions[release]
+		if ok {
+			targetVersions.Delete(release)
+		}
+	}
+	for _, clone := range clones {
+		clonePRs, err := client.GetExternalBugPRsOnBug(clone.ID)
+		// Remove target releases which already have clones
+		for _, release := range clone.TargetRelease {
+			_, ok := targetVersions[release]
+			if ok {
+				targetVersions.Delete(release)
+			}
+		}
+
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("Bug#%d - error occured while retreiving list of PRs : %v", clone.ID, err))
+			if wpErr != nil {
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
+				return nil, wpErr
+			}
+			return nil, err
+		}
+		clone.PRs = clonePRs
+	}
+	wrpr := ClonesTemplateData{
+		Bug:          bug,
+		Clones:       clones,
+		Parent:       parent,
+		PRs:          prs,
+		CloneTargets: targetVersions.List(),
+	}
+	return &wrpr, nil
+}
+
+// ClonesHandler acts as a router for the RESTish calls to clones
+func ClonesHandler(client bugzilla.Client, allTargetVersions sets.String) HandlerFuncWithErrorReturn {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusBadRequest)
+			wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
+			if wpErr != nil {
+				http.Error(w, "Error building page.", http.StatusInternalServerError)
+				return wpErr
+			}
+			return fmt.Errorf("Bad request - clones can handle only GET requests")
+		}
+		return GetClonesHandler(client, allTargetVersions, w, r)
+	}
+}
+
 // GetClonesHandler returns an HTML page with detais about the bug and its clones
-func GetClonesHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+func GetClonesHandler(client bugzilla.Client, allTargetVersions sets.String, w http.ResponseWriter, req *http.Request) error {
+	bugIDStr := req.URL.Query().Get(BugIDQuery)
+	if bugIDStr == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		wpErr := writePage(w, "Error.", errorTemplate, fmt.Sprintf("%s - query incorrect", BugIDQuery))
```

Comment:
Can we reword this to `missing mandatory query arg: BugIDQuery`? The current error message is confusing at best